lib/sales_analyst.rb

```diff
@@ -107,6 +108,8 @@ def merchants_with_high_item_count
     end
     merchant_ids_with_high_item_count.map do |merchant_id|
       @merchants_repo.find_by_id(merchant_id)
+      #where you are iterating more than once in one method
```

Comment:
Thank you for these refactor comments, I will look into reducing nested iteration in this class